app/account/page.jsx

```diff
@@ -46,29 +48,68 @@ function a11yProps(index) {
 
 export default function BasicTabs() {
   const [currentTab, setCurrentTab] = React.useState(0);
+  const [purchases, setPurchases] = React.useState([]);
   const { user, userData, setUserData } = useUserContext();
   const router = useRouter();
+  let groupedByDate;
+  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
 
-  const fetchUserPath = '../api/userData';
-  const action = 'getuser';
-  const collection = 'userdata';
-  const document = user?.uid.toString() || null;
-  const requestPath = `${fetchUserPath}?action=${action}&collection=${collection}&document=${document}`
-  const { data, error, isLoading } = useSWR(requestPath, customGetter);
   useEffect(() => {
-
     if (user === null) {
-      router.push('/signin')
-    }
+      router.push('/signin');
```

Comment:
in the [category] page.jsx file, you also have some user signed in logic where you route to the registration page. I'm assuming you'll likely have this logic in other pages, should we create a reusable hook to have that logic modularized? this is up to you.